### PR TITLE
[Merged by Bors] - do not impl Component for Task

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- `Task` are `Component`.
- They should not.

## Solution

- Remove the impl, and update the example to show a wrapper.

#4052 for reference

### CODE REVIEW

**Comment by DJMcNab** [comment]
You're right, I did link the wrong PR in https://github.com/bevyengine/bevy/pull/4052#pullrequestreview-894474081; the link should have been https://github.com/bevyengine/bevy/pull/2975.

**Comment by mockersf** [comment]
Oh I missed that was part of that PR!

**Comment by mockersf** [comment]
This is better as a standalone PR as it seems #2975 may take some time before it could be merged

**Review by alice-i-cecile — APPROVED** [review]

**Review by hymm — APPROVED** [review]

**Review by MrGVSV — APPROVED** [review]

File: examples/async_tasks/async_compute.rs [hunk 1]
```diff
@@ -41,6 +41,9 @@ fn add_assets(
     commands.insert_resource(BoxMaterialHandle(box_material_handle));
 }
 
+#[derive(Component)]
+struct ComputeTransform(Task<Transform>);
```
Comment (pubrrr) [inline examples/async_tasks/async_compute.rs:45]: FYI, I also did this is #4102.

**Comment by IceSentry** [comment]
Should this be considered a breaking change?

**Pull request closed without merging**